Add initial version of 3scale alert SOPs included on 3scale v2.9 (#9)
3scale/3scale-Operations: sops/alerts/zync_que_failed_job_count_high.adoc

:toc:
:toc-placement!:

= ThreescaleZyncQueFailedJobCountHigh

toc::[]

== Description

* This alert will trigger if the number of `zync-que` **failed** jobs is greater than a given threshold

> * **Failed** jobs are the ones that have failed at least once

Author: ```suggestion
* **Failed** jobs are the ones that failed at least once, did not run out of attempts to retry and therefore are scheduled for retry any time soon
```

== Troubleshooting

> * Check if `zync-que` pods might be failing (so no consuming any job from postgres)

Author: ```suggestion
* Check the logs at `zync-que` pods to see the reason of the failures
```

> * Check if `zync-que` pods might not be consuming jobs at the required speed (too much cpu usage causing slowliness?)
> - You may need to consider scaling horinzontally `zync-que` deployment
> - You may need to consider increasing `zync-que` deployment resources requests/limits

Author: ```suggestion
```

> * Check if `zync` might not be be able to deal with the incoming traffic, generating too many jobs at postgres (maybe `zync` is being attacked?)

Author: ```suggestion
```

> * Check if postgres is having some issues

Author: ```suggestion
```
